### ISSUE TITLE
template friendly getters?

### ISSUE THREAD

**Issue by schleichardt** [sphereio/sphere-jvm-sdk#226]
```
//could be added
default String sku() {
  return getSku().orElse("");
}

Optional<String> getSku();//current situation




default LocalizedStrings description() {
  return getDescription().orElse(EMPTY_LOCALIZED_STRINGS);
}

Optional<LocalizedStrings> getDescription();
```

@lauraluiz 
@nkuehn 
What do you think?

**Comment by nkuehn** [comment]
+1 a lot - but not sure what's the most elegant way.

That's probably the kind of stuff done in Project specific helper libraries.  Coolest would be something that allows to work like a safe-dereference operator because you often go through a chain of accessors. Dream on, this is Java. 

Sticking to your case, two variants:

Overload the getter (but that would require to always pass the fallback explicitly)
```java
Optional<String> getSku(){
  return this.sku;
}
String getSku(String fallback){
  return getSku().orElse(fallback);
}
```
alternative naming pattern (more explicit)
```java
Optional<String> getSku(){
  return this.sku;
}
String safeSku(String fallback){
  return getSku().orElse(fallback);
}
```

**Comment by schleichardt** [comment]
No this pattern I wont support, since Optional provides this fuctionality

**Comment by nkuehn** [comment]
"this" is probably the first one, right?  I should have read this  http://www.oracle.com/technetwork/articles/java/java8-optional-2175753.html first :-(  

If the typical pattern in a template is the kind of flatMap() cascade shown there, I'm not sure what we're saving with the proposa.  
tl;dr: I drop out of the discussion or need more examples. Too bad a Java 8.

**Comment by lauraluiz** [comment]
I think we should leave this specific cases to the project developers. We are providing an excellent object with the Optional to work with in templates. If they need more control they can implement Page classes that extract the information exactly as they need it for the templates (I really liked the idea).
